### PR TITLE
Christoph's changes to EXTRA_FIM

### PR DESCRIPTION
**Pull request by freyso**
Things I did
- all the code changes I had done so far, as discussed this afternoon
- introduced the -k vector fix in the sx_waves_reader class (happens in get_kvec)
- format "black" for all my new python files, and all previous files in EXTRA_FIM
  btw: on the command line: 
  ` python -m black <files>`
- replaced all np.einsum ways of computing plane averages by np.mean
- added my example notebooks in _Examples/_, **these won't run without my data**

Things I did not do:
- touch _data_utils/post_processing.py_
- rename any parameters
- try to be consistent with units
- reorganize code distribution over files

Please don't be shy in criticising my new code, I am not an expert in python programming.

### CODE REVIEW

File: .vscode/settings.json
Comment (skatnagallu) [inline .vscode/settings.json:1]: These are mostly settings for vscode. I guess no harm in leaving them int

File: EXTRA_FIM/main.py [hunk 1]
```diff
@@ -155,7 +159,7 @@ def iso_contour( self,gk_1,gk_2,k_vec,cutoff,izstart_min,limit):
                         break
                     else:
```
Comment (skatnagallu) [inline EXTRA_FIM/main.py:160]: we can use `elif i==izstart_min: `

File: data_utils/post_processing.py
Comment (skatnagallu) [inline data_utils/post_processing.py:1]: This I guess becomes redundant. We can fix this later.

**Review by skatnagallu — APPROVED** [review]
I think this looks good to me. But I haven't checked if the branch works exclusively on the cluster.  

Can be merged.

**Review by Bhatt993 — APPROVED** [review]

**Pull request merged**